pkg/experiment/ingester/segment.go

```diff
@@ -216,7 +216,7 @@ func (sw *segmentsWriter) newSegment(sh *shard, sk shardKey, sl log.Logger) *seg
 func (s *segment) flush(ctx context.Context) (err error) {
 	span, ctx := opentracing.StartSpanFromContext(ctx, "segment.flush", opentracing.Tags{
 		"block_id": s.ulid.String(),
-		"datasets": len(s.heads),
+		"datasets": len(s.datasets),
```

Comment:
Is this rename needed for the change?

Reply:
Not at all. It was a rename we did with Anton when review this, but to keep minimal changes we can leave it as `heads`. 

PS: I realised I didn't rename `headForIngest`.